packages/gatsby-cli/cli.js

```diff
@@ -1,3 +1,34 @@
 #!/usr/bin/env node
 
+const semver = require("semver")
+const commonTags = require("common-tags")
+
+const MIN_NODE_VERSION = `10.13.0`
+
+const version = process.version
+
+if (
+  !semver.satisfies(version, `>=${MIN_NODE_VERSION}`, {
+    includePrerelease: true,
+  })
+) {
+  console.error(
+    commonTags.stripIndent(`
+      Gatsby requires Node.js ${MIN_NODE_VERSION} or higher (you have ${version}).
+      Upgrade Node to the latest stable release: https://gatsby.dev/upgrading-node-js
+    `)
+  )
+  process.exit()
```

Comment:
This should be non-success error code I think
```suggestion
  process.exit(1)
```